cpmpy/expressions/globalconstraints.py

```diff
@@ -223,6 +223,12 @@ def __init__(self, *args):
         flatargs = flatlist(args)
         if any(is_boolexpr(arg) for arg in flatargs):
             raise TypeError("Circuit global constraint only takes arithmetic arguments: {}".format(flatargs))
+        lbs, ubs = zip(*[get_bounds(arg) for arg in flatargs])
+        lb = min(lbs)
+        ub = max(ubs)
+        if lb < 0 or ub >= len(flatargs):
+            raise TypeError("Circuit global constraint is only defined if all bounds are between 0 and the length of the array.")
```

Comment:
I wouldn't call it a type error, but rather a PartialFunctionError or something?  I think we even have one already

Reply:
Maybe make the error message more explicit too? 
Smt like: `elements in the array must have bounds 0 <= x <= len(flatargs) but got bounds [{lb},{ub}] for arg ...`